### PR TITLE
chore(release): Improve release workflow caching

### PR DESCRIPTION
**Pull request by erezrokah**
#### Summary

Related to https://github.com/cloudquery/cloudquery/issues/2757.

At the moment all our workflows share the same cache key logic that is a composition of the OS, Go version, and a hash of the `go.sum` file of the specific component the workflow runs against.
This means that our release workflows (validate release and Go releaser) share the same cache as our testing workflows.
Which means a pull request can override the cache of the release workflows.

I think we would benefit from having a dedicated cache just for releases (which this PR does).

**TLDR**: This PR creates a dedicated cache for the `validate release` workflow and `release plugin/cli` workflows (a cache per component).

You can see an example here:
First run of AWS validate release is ~20 minutes:
https://github.com/cloudquery/cloudquery/actions/runs/3259211192/jobs/5351872391
image
image
Second run of AWS validate release is ~2 minutes (10x faster):
https://github.com/cloudquery/cloudquery/actions/runs/3259211192/jobs/5352039966
image

Still need to test this once merged to see if there's an improvement

### CODE REVIEW

**Review by yevgenypats — APPROVED** [review]

**Comment by erezrokah** [comment]
PLEASE DON'T MERGE, I'll MERGE IT ONCE I'M AVAILABLE TO FIX ISSUES IF THIS BREAKS ANYTHING WITH RELEASES
